fixpatches/src/commonMain/kotlin/com/saveourtool/sarifutils/net/UrilUtils.kt

```diff
@@ -0,0 +1,176 @@
+@file:JvmName("UriUtils")
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.sarifutils.net
+
+import com.saveourtool.okio.BACKSLASH
+import com.saveourtool.okio.SLASH
+import com.saveourtool.okio.URI_UNC_PATH_PREFIX
+import com.saveourtool.okio.Uri
+import com.saveourtool.okio.backslashify
+import com.saveourtool.okio.slashify
+import com.saveourtool.okio.toLocalPath
+import com.saveourtool.system.OsFamily
+import okio.Path
+import okio.Path.Companion.toPath
+import kotlin.jvm.JvmName
+
+private typealias UriPath = CharSequence
+
+/**
+ * Converts this URI to a local path.
+ *
+ * Puts more effort into the conversion than [Uri.toLocalPath],
+ * covering some extra corner cases.
+ *
+ * @return the local path created from this `file://` or UNC URI.
+ * @throws IllegalArgumentException if the URI contains an absolute path from
+ *   the different platform (e.g.: a Windows path while the current OS is UNIX).
+ * @see Uri.toLocalPath
+ */
+@Suppress(
+    "CyclomaticComplexMethod",
+    "NestedBlockDepth",
+    "TOO_LONG_FUNCTION",
+)
+internal fun Uri.toLocalPathExt(): Path =
+        when {
+            isAbsoluteWindowsPath() -> {
+                val localPath = "$scheme:${path ?: schemeSpecificPart}"
+
+                localPath.requireOsIsWindows()
+
+                localPath.backslashify().toPath()
+            }
+
+            isAbsolute -> when (val path = path) {
+                /*
+                 * When a URI is opaque, its path is `null`.
+                 */
+                null -> schemeSpecificPart.toPath()
+
+                else -> {
+                    @Suppress("WHEN_WITHOUT_ELSE")
+                    when {
+                        /*
+                         * This is not 100% correct, as a
+                         * normalized UNC URI is indistinguishable
+                         * from a URI that holds an absolute
+                         * UNIX path, e.g.:
+                         * `file:/WSL$/Debian/etc/passwd`.
+                         */
+                        path.isAbsoluteUnixPath() && authority == null -> path.requireOsIsUnix()
+                        path.isAbsoluteWindowsPath() -> path.requireOsIsWindows()
+                        path.isUncPath() -> path.requireOsIsWindows()
+                    }
+
+                    toLocalPath()
+                }
+            }
+
+            else -> {
+                val path = path
```

Comment:
Why do we need this?

Reply:
Referenced from #40, marking as resolved for now.